### PR TITLE
Register Visitor Feature

### PR DESCRIPTION
**Pull request by A23W10**
### **What does this PR do?**
- Get visitor's information and store it in the visitor's table on the database

### **Description of tasks:**
- Create visitor's table
- Get the visitors information and store it onto the database

### **How can you test this manually**
- Using Postman, and the link [localhost:4002/api/visitors/register](url), the visitor can enter their details which are then returned indicating the fields stored onto the database

### **Screenshot(s)**
_**Visitors Table in Database**_
VisitorsTable

_**Postman Request**_
PostmanRequest

_**Postman Response**_
PostmanResponse

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (lytes20) [inline .vscode/settings.json:1]: Could you please remove this file from your commit

File: .gitignore [hunk 2]
```diff
@@ -34,3 +34,5 @@ lerna-debug.log*
 !.vscode/extensions.json
 .env.prod
 *.prod
+
+.env
```
Comment (lytes20) [inline .gitignore:38]: Could you please add a new at the end of this file

File: src/visitors/enums/whLocation.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    WHKBY = 'WHKBY',    //Kibuye
+    WHKNG = 'WHKNG',    //Kungu
+    WHMAK = 'WHMAK',    //Makerere
+    WHMBALE = 'WHMBALE',//Mbale
+    WHMBR = 'WHMBR',    //Mbarara
+    WHMKN = 'WHMKN',    //Mukono
+    WHNAI = 'WHNAI',    //Nairobi
+    WHNLY = 'WHNLY'     //Naalya
+  }
+
```
Comment (lytes20) [inline src/visitors/enums/whLocation.ts:23]: Could you please add a new line at the end of this file

File: src/visitors/dto/visitorDto.dto.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { IsAlpha, IsEnum, IsNotEmpty, IsPhoneNumber, IsString } from "class-validator";
+import { Gender } from "src/crm/enums/gender";
+import { WHLocation } from "../enums/whLocation";
+
+//DTO to return the visitor information
```
Comment (lytes20) [inline src/visitors/dto/visitorDto.dto.ts:5]: I think this comment should: to validate visitor information, could you please edit it

**Comment by lytes20** [comment]
Could you run your tests locally, if they pass, it's okay, commit them, if they don't remove them

**Review by MariamNaiga — APPROVED** [review]

**Review by lytes20 — APPROVED** [review]

**Comment by lytes20** [comment]
LGTM
cc @ekastimo

**Pull request closed without merging**